mijia-homie/src/main.rs

```diff
@@ -396,42 +417,64 @@ async fn connect_sensor_with_id(
     Ok(())
 }
 
-async fn connect_and_subscribe_sensor_or_disconnect<'a>(
+// TODO: Does this already exist somewhere?
+/// Return the first Ok element of the stream, or a vector of all errors if they all fail.
+async fn first_ok<V, E>(stream: impl Stream<Item = Result<V, E>>) -> Result<V, Vec<E>> {
```

Comment:
kill this and use a for loop below instead.

Reply:
Done, though I'm sad about it.